msticpy/data/uploaders/loganalytics_uploader.py

```diff
@@ -0,0 +1,238 @@
+                Check Workspace ID and key""",
+                title="Data Upload Failed",
+            )
+
+    def upload_df(self, data: pd.DataFrame, table_name: Any, **kwargs):
+        """
+        Upload a pandas DataFrame to Log Analytics.
+
+        Parameters
+        ----------
+        data : pd.DataFrame
+            Pandas DataFrame to upload.
+        table_name : str
+            Custom table name to upload the data to.
+
+        """
+        events = []
+        for row in data.iterrows():
+            events.append(row[1].astype(str).to_dict())
+            # Due to 30MB limit if data is larger than 25Mb upload that chunk then continue
+            if sys.getsizeof(json.dumps(events)) > 26214400:
+                if self._debug is True:
+                    print("Data larger than 25MB spliting data requests.")
+                body = json.dumps(events)
+                self._post_data(body, table_name)
+                events = []
+
+        if events:
+            body = json.dumps(events)
+            self._post_data(body, table_name)
+
+        if self._debug:
+            print(f"Upload to {table_name} complete")
+
+    def upload_file(
+        self, file_path: str, table_name: str = None, delim: str = ",", **kwargs,
+    ):
+        """
+        Upload a seperated value file to Log Analytics.
+
+        Parameters
+        ----------
+        file_path : str
+            Path to file to upload.
+        table_name : str
+            Table name to upload data to.
+        delim : str, optional
+            Value seperator used by the file, by default ","
+
+        """
+        path = Path(file_path)
+        data = pd.read_csv(path, delimiter=delim)
+        if not table_name:
+            table_name = str(path).split("\\")[-1].split(".")[0]
+        self.upload_df(data, table_name)
+
+    def upload_folder(
+        self, folder_path: str, table_name: str = None, delim: str = ",", **kwargs
+    ):
+        """
+        Upload all files in a folder to Log Analytics.
+
+        Parameters
+        ----------
+        folder_path : str
+            Folder path to upload.
+        table_name : str, optional
+            Table name to upload all files to, by default None.
+        delim : str, optional
+            Sperator used in files in target folder, by default ",".
+
+        """
+        if delim != ",":
+            ext = "*"
+        else:
+            ext = "*.csv"
+        t_name = bool(table_name)
+        input_files = Path(folder_path).glob(ext)
+        # pylint: disable=unnecessary-comprehension
+        input_files = [
+            path for path in input_files  # type: ignore
+        ]
+        # pylint: enable=unnecessary-comprehension
+        progress = tqdm(total=len(list(input_files)), desc="Files", position=0)
+        for path in input_files:
+            data = pd.read_csv(path, delimiter=delim)
+            if t_name is False:
+                table_name = str(path).split("\\")[-1].split(".")[0]
```

Comment:
Should use Path from pathlib. This handles different OS separators. You can also get the file name minus extension (stem? maybe)